src/wallet.js

```diff
@@ -723,6 +723,84 @@ const wallet = {
     }
   },
 
+  /**
+   * Validate old password and change it for the new one
+   *
+   * @param {string} oldPassword
+   * @param {string} newPassword
+   *
+   * @return {boolean} true if the PIN was successfully changed
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  changePassword(oldPassword, newPassword) {
+    if (this.isFromXPub()) {
+        throw WalletFromXPubGuard('changePassword');
+    }
+
+    const isCorrect = this.isPasswordCorrect(oldPassword);
+    if (!isCorrect) {
+      return false;
+    }
+
+    // Get new password hash
+    const newHash = this.hashPassword(newPassword);
+    // Update new Password data in storage
+    const accessData = this.getWalletAccessData();
+    accessData['hashPasswd'] = newHash.key.toString();
+    accessData['saltPasswd'] = newHash.salt;
+
+    // Get and update seed encrypted with PIN
+    const decryptedData = this.decryptData(accessData.words, oldPassword);
+    const encryptedData = this.encryptData(decryptedData, newPassword);
+    accessData['words'] = encryptedData.encrypted.toString();
+
+    this.setWalletAccessData(accessData);
+
+    return true;
+  },
+
+  /**
+   * @return {[boolean, string]} Tuple <success:boolean, pin:string> with the result
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  guessPin() {
+    const CryptoJS = require('crypto-js');
+    const accessData = this.getWalletAccessData();
+    const data = accessData.words;
+
+    const decrypt = (data, pin) => CryptoJS
+      .AES
+      .decrypt(data, pin)
+      .toString(CryptoJS.enc.Utf8);
+
+    let pinFound = -1;
+    for (let i = 0; i <= 999999; i++) {
```

Comment:
Should we block the main loop? I think we should do something like this:

```js
guessPartial(begin, end) {
  :
  for (let i = begin; i <= end; i++) {
    :
  }
  :
}

guessScheduler(begin, step, cb) {
  const end = begin + step;
  ret = guessPartial(begin, end);
  if (ret[0] == false) {
    if (end < 999999) {
      setTimeout(guessScheduler, 0, end + 1, step, cb);
    }
  } else {
    cb(pin);
  }
}

guessPin(cb) {
  guessScheduler(0, 1000, cb);
}

Reply:
Maybe we should call `cb(...)` multiple times so the caller can update the interface with progress.

Reply:
@msbrogli This was moved to the hathor-wallet-mobile repo, can you check it there?